L1Trigger/L1TTrackMatch/plugins/L1FastTrackingJetProducer.cc

```diff
@@ -0,0 +1,257 @@
+        else if (detId.subdetId() == StripSubdetector::TID && tTopo.tidRing(detId) <= 9)
+          tmp_isPS = true;
+      }
+      if (tmp_isPS)
+        trk_nPS++;
+    }
+    if (trk_nPS < trkNPSStubMin_)
+      continue;
+    if (fabs(recoVtx - trk_z0) > deltaZ0Cut_)
+      continue;
+    if (!iEvent.isRealData()) {
+      edm::Ptr<TTTrack<Ref_Phase2TrackerDigi_>> trk_ptr(TTTrackHandle, this_l1track);
+
+      if (!(MCTrkAssociation->isGenuine(trk_ptr)) && selectTrkMatchGenTight_)
+        continue;
+      if (!(MCTrkAssociation->isLooselyGenuine(trk_ptr) || MCTrkAssociation->isGenuine(trk_ptr)) &&
+          selectTrkMatchGenLoose_)
+        continue;
+      if (!(MCTrkAssociation->isLooselyGenuine(trk_ptr) || MCTrkAssociation->isGenuine(trk_ptr) ||
+            MCTrkAssociation->isCombinatoric(trk_ptr)) &&
+          selectTrkMatchGenOrPU_)
+        continue;
+    }
+
+    fastjet::PseudoJet psuedoJet(iterL1Track->momentum().x(),
+                                 iterL1Track->momentum().y(),
+                                 iterL1Track->momentum().z(),
+                                 iterL1Track->momentum().mag());
+    JetInputs.push_back(psuedoJet);                     // input tracks for clustering
+    JetInputs.back().set_user_index(this_l1track - 1);  // save track index in the collection
+  }                                                     // end loop over tracks
+
+  fastjet::ClusterSequence cs(JetInputs, jet_def);  // define the output jet collection
+  std::vector<fastjet::PseudoJet> JetOutputs =
+      fastjet::sorted_by_pt(cs.inclusive_jets(0));  // output jet collection, pT-ordered
+
+  for (unsigned int ijet = 0; ijet < JetOutputs.size(); ++ijet) {
+    math::XYZTLorentzVector jetP4(
+        JetOutputs[ijet].px(), JetOutputs[ijet].py(), JetOutputs[ijet].pz(), JetOutputs[ijet].modp());
+    float sumpt = 0;
+    float avgZ = 0;
+    std::vector<edm::Ptr<L1TTTrackType>> L1TrackPtrs;
+    std::vector<fastjet::PseudoJet> fjConstituents = fastjet::sorted_by_pt(cs.constituents(JetOutputs[ijet]));
+
+    for (unsigned int i = 0; i < fjConstituents.size(); ++i) {
+      auto index = fjConstituents[i].user_index();
+      edm::Ptr<L1TTTrackType> trkPtr(TTTrackHandle, index);
+      L1TrackPtrs.push_back(trkPtr);  // L1Tracks in the jet
+      sumpt = sumpt + trkPtr->momentum().perp();
+      avgZ = avgZ + trkPtr->momentum().perp() * trkPtr->z0();
+    }
+    avgZ = avgZ / sumpt;
+    edm::Ref<JetBxCollection> jetRef;
+    TkJet trkJet(jetP4, jetRef, L1TrackPtrs, avgZ);
+    L1FastTrackingJets->push_back(trkJet);
+  }  //end loop over Jet Outputs
+
+  if (displaced_)
+    iEvent.put(std::move(L1FastTrackingJets), "L1FastTrackingJetsExtended");
+  else
+    iEvent.put(std::move(L1FastTrackingJets), "L1FastTrackingJets");
+}
+
+void L1FastTrackingJetProducer::beginJob() {}
+
+void L1FastTrackingJetProducer::endJob() {}
+
+void L1FastTrackingJetProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
```

Comment:
Suggest dropping the cfi file in favour of adding the fillDescriptions class. You can generate the pset description using 
```
edmPythonConfigToCppValidation L1FastTrackingJetProducer_cfi.py 
```